adp/rbac/roles.py

```
@@ -0,0 +1,477 @@
+            entity_id: Entity to assign role to
+            role_name: Name of role to assign
+            assigned_by: Entity making the assignment
+            expires_at: Optional expiration time
+            
+        Returns:
+            RoleAssignment object
+        """
+        # Verify role exists
+        role = self._hierarchy.get_role(role_name)
+        if not role:
+            raise ValueError(f"Role {role_name} does not exist")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)